File: Homework0/Homework0 - c++/Homework0.cpp
/*
	CSC345 - Homework 0 - C++
	By: Brent Turner
	09/02/17
*/
#include "stdafx.h"
#include <iostream>
using namespace std;

//Pass by Reference given (prototypes)
int square(int &x);
int twice(int &x);
int once(int &x);

int main()
{
	int x = 3;
	int y = square(x) + twice(x) + once(x); //y == 45, not 18
	cout << y << endl; //prints 45 not 18
	_sleep(1000);
	return 0;
}

/*Square function x * x */
int square(int &x)
{
	x = x * x;
	return x;
}
/*2 * x function */
int twice(int &x)
{
	x = x * 2;
	return x;
}

/*just x function*/
int once(int &x)
{
	return x;
}

File: Homework0/Java/src/Homework0.java
/*
 * CSC 345 - Homework 0
 * By: Brent Turner
 * Homework0.java
 * 09/04/17
 */
public class Homework0 
{
	public static void main(String[] args) 
	{
		Int x = new Int(3);
		int y  = square(x) + twice(x) + once(x); //y == 45, not 18
		System.out.println(y);					//prints 45, not 18
	}
	
	/*
	 * square root
	 * @param x
	 */
	private static int square(Int x) //x = x*x
	{
		x.setInt(x.getInt() * x.getInt());
		return x.getInt();
	}
	
	/*
	 * multiply by 2
	 * @param x
	 */
	private static int twice(Int x) // x = x * 2
	{
		x.setInt(x.getInt() * 2);
		return x.getInt();
	}
	
	/*
	 * just the value
	 * @param x
	 */
	private static int once(Int x) // x = x
	{
		return x.getInt();
	}
}
